stdlib/public/core/StringBridge.swift

```diff
@@ -413,10 +413,42 @@ private func _getCocoaStringPointer(
   return .none
 }
 
+#if arch(arm64)
+//11000000..payload..111
+private var constantTagMask:UInt {
+  0b1111_1111_1000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0111
+}
+private var expectedConstantTagValue:UInt {
+  0b1100_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0000_0111
+}
+#endif
+
+@inline(__always)
+private func formConstantTaggedCocoaString(
+  untaggedCocoa: _CocoaString
+) -> AnyObject? {
+#if !arch(arm64)
+  return nil
+#else
+
+  let constantPtr:UnsafeRawPointer = Builtin.reinterpretCast(untaggedCocoa)
+
+  // Check if what we're pointing to is actually a valid tagged constant
+  guard _swift_stdlib_dyld_is_objc_constant_string(constantPtr) == 1 else {
+    return nil
+  }
+
+  let retaggedPointer = UInt(bitPattern: constantPtr) | expectedConstantTagValue
+
+  return unsafeBitCast(retaggedPointer, to: AnyObject.self)
+#endif
```

Comment:
This still seems a little sketchy without at least a `defer { _fixLifetime(untaggedCocoa) }`. Make sure to check with @atrick or @eeckstein.

In practice, would ref counting always be disabled here? How are the retain/release counts balanced?

Reply:
These objects are in immutable memory, so can't be refcounted

Reply:
👍